### PR TITLE
Custom rolling index pattern name separator

### PR DESCRIPTION
**Pull request by cuneytcalishkan**
Hello,
here is a PR for a custom rolling index pattern name separator for issue #37 
What do you think?
Cheers,
Cuneyt

### CODE REVIEW

File: log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatter.java [hunk 1]
```diff
@@ -65,11 +64,12 @@
     private int DEFAULT_BUFFER_SIZE = 32;
     private long currentFileTime;
 
-    protected RollingIndexNameFormatter(String indexName, String pattern, long initTimeInMillis, TimeZone timeZone) {
+    protected RollingIndexNameFormatter(String indexName, String pattern, long initTimeInMillis, TimeZone timeZone, String separator) {
```
Comment (rfoltyns) [inline log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatter.java]: This change will break extending classes. Could you add a separate constructor with new parameter, please? Current constructor should invoke this with DEFAULT_SEPARATOR.

File: log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatter.java [hunk 2]
```diff
@@ -9,9 +9,7 @@
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
- *
```
Comment (rfoltyns) [inline log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatter.java]: Out of scope

File: log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatter.java [hunk 3]
```diff
@@ -9,9 +9,7 @@
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
- * 
  *      http://www.apache.org/licenses/LICENSE-2.0
- *
```
Comment (rfoltyns) [inline log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatter.java]: Out of scope

File: log4j2-elasticsearch-core/src/test/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatterTest.java [hunk 4]
```diff
@@ -9,9 +9,7 @@
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
- *
```
Comment (rfoltyns) [inline log4j2-elasticsearch-core/src/test/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatterTest.java]: Out of scope

File: log4j2-elasticsearch-core/src/test/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatterTest.java [hunk 5]
```diff
@@ -9,9 +9,7 @@
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
- * 
  *      http://www.apache.org/licenses/LICENSE-2.0
- *
```
Comment (rfoltyns) [inline log4j2-elasticsearch-core/src/test/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatterTest.java]: Out of scope

File: log4j2-elasticsearch-core/src/test/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatterTest.java [hunk 6]
```diff
@@ -54,8 +52,8 @@
 
     private static long getTestTimeInMillis() {
         return LocalDateTime.of(2017, 12, 20, 23, 54, 0, 0)
-                .atZone(ZoneId.of(RollingIndexNameFormatter.Builder.DEFAULT_TIME_ZONE))
```
Comment (rfoltyns) [inline log4j2-elasticsearch-core/src/test/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatterTest.java]: Out of scope

File: log4j2-elasticsearch-core/src/test/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatterTest.java [hunk 7]
```diff
@@ -54,8 +52,8 @@
...
-                .toInstant().toEpochMilli();
```
Comment (rfoltyns) [inline log4j2-elasticsearch-core/src/test/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatterTest.java]: Out of scope

File: log4j2-elasticsearch-core/src/test/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatterTest.java [hunk 8]
```diff
@@ -81,7 +79,7 @@ public void startsWithoutExceptionsIfSetupIsCorrect() {
     }
 
     @Test(expected = ConfigurationException.class)
-    public void builderThrowsExceptioWhenIndexNameIsNull() {
+    public void builderThrowsExceptionWhenIndexNameIsNull() {
```
Comment (rfoltyns) [inline log4j2-elasticsearch-core/src/test/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatterTest.java]: Out of scope

File: log4j2-elasticsearch-core/src/test/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatterTest.java [hunk 9]
```diff
@@ -93,7 +91,7 @@ public void builderThrowsExceptioWhenIndexNameIsNull() {
 
 
     @Test(expected = ConfigurationException.class)
-    public void builderThrowsExceptioWhenPatternIsNull() {
+    public void builderThrowsExceptionWhenPatternIsNull() {
```
Comment (rfoltyns) [inline log4j2-elasticsearch-core/src/test/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatterTest.java]: Out of scope

File: log4j2-elasticsearch-core/src/test/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatterTest.java [hunk 10]
```diff
@@ -202,14 +217,14 @@ public void run() {
     public void concurrencyTest() throws InterruptedException {
 
         // given
-//        TestFormatter.TEST_PATTERN_PROCESSOR = Mockito.spy(new PatternProcessor("%d{" + DATE_PATTERN_WITH_MINUTES + "}"));
+        // TestFormatter.TEST_PATTERN_PROCESSOR = Mockito.spy(new PatternProcessor("%d{" + DATE_PATTERN_WITH_MINUTES + "}"));
```
Comment (rfoltyns) [inline log4j2-elasticsearch-core/src/test/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatterTest.java]: Out of scope

File: log4j2-elasticsearch-core/src/test/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatterTest.java [hunk 11]
```diff
@@ -202,14 +217,14 @@ public void run() {
...
 
         for (int ii = 0; ii < 100; ii++) {
             IndexNameFormatter formatter = createRollingIndexNameFormatterBuilder().build();
             runSingleConcurrencyTest(formatter, 20);
         }
 
-//        Mockito.verify(TestFormatter.TEST_PATTERN_PROCESSOR, Mockito.times(100)).getNextTime(Mockito.any(long.class), Mockito.eq(1), Mockito.eq(false));
+        // Mockito.verify(TestFormatter.TEST_PATTERN_PROCESSOR, Mockito.times(100)).getNextTime(Mockito.any(long.class), Mockito.eq(1), Mockito.eq(false));
```
Comment (rfoltyns) [inline log4j2-elasticsearch-core/src/test/java/org/appenders/log4j2/elasticsearch/RollingIndexNameFormatterTest.java]: Out of scope

**Comment by cuneytcalishkan** [comment]
I will create a fresh new PR with all the remarks taken into account. Please just ignore this one.

**Pull request closed without merging**